.scripts/summarize_progress/main.py

```diff
@@ -0,0 +1,138 @@
+import re
+import polib
+import glob
+import datetime
+import requests
+
+
+def entry_check(pofile) -> str:
+    lines_tranlated = len(pofile.translated_entries())
+    lines_untranlated = len(pofile.untranslated_entries())
+
+    if lines_tranlated == 0:
+        result = "❌"
+    elif lines_untranlated == 0:
+        result = "✅"
+    else:
+        lines_all = lines_tranlated + lines_untranlated
+        progress = lines_tranlated / lines_all
+        progress_percentage = round(progress * 100, 2)
+        result = f"Ongoing, {str(progress_percentage)} %"
+
+    return result
+
+
+def get_open_issues_count() -> int:
+    url = f"https://api.github.com/search/issues?q=repo:python/python-docs-zh-tw+type:issue+state:open"
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    r = requests.get(url=url, headers=headers)
+    result = r.json()
+
+    return result["total_count"]
+
+
+def get_github_issues() -> list:
+    NUMBER_OF_ISSUES = get_open_issues_count()
+
+    url = f"https://api.github.com/search/issues?q=repo:python/python-docs-zh-tw+type:issue+state:open&per_page={NUMBER_OF_ISSUES}"
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    r = requests.get(url=url, headers=headers)
+    result = r.json()
+
+    result_list = []
+    for issue in result["items"]:
+        title_segments = issue["title"].split()
+
+        if len(title_segments) < 2:
+            continue
+        if title_segments[0] != "翻譯" and title_segments[0].lower() != "translate":
+            continue
+        if issue["assignee"] is None:
+            continue
+
+        filename = title_segments[1].strip("`")
+
+        if re.fullmatch("[a-zA-z-]+/[a-zA-Z0-9._-]+.po", filename):
+            filename_segments = filename.split("/")
+            # print(filename_segments)
+        elif re.fullmatch("[a-zA-z-]+/[a-zA-Z0-9._-]+", filename):
+            filename_segments = filename.split("/")
+            filename_segments[1] += ".po"
+        else:
+            continue
+
+        result_list.append([filename_segments, issue["assignee"]["login"]])
+
+    return result_list
```

Comment:
改寫供參考（但沒測試過XD）
- nit: 會希望盡量不要用到 split，就盡量用 re 取代

```suggestion
    for issue in result["items"]:
        if issue["assignee"] is None:
            continue

        title = issue["title"]
        if "翻譯" not in title and "translate" not in title.lower():
            continue

        match = re.search("(?P<dirname>[a-zA-z-]+)/(?P<filename>[a-zA-Z0-9._-]+(.po)?)", title)
        if not match:
            continue
        
        dirname, filename = match.group('dirname', 'filename')
        if not filename.endswith('.po'):
            filename += '.po'

        result_list.append(((dirname, filename), issue["assignee"]["login"]))

    return result_list
```

Reply:
感謝！這邊測試過了沒問題！